ydb/core/kqp/provider/read_attributes_utils.cpp

```diff
@@ -215,39 +216,13 @@ void ReplaceReadAttributes(TExprNode& node,
     TraverseReadAttributes(visitor, node, ctx);
 }
 
-static Ydb::Type CreateYdbType(const NKikimr::NScheme::TTypeInfo& typeInfo, bool notNull) {
-    Ydb::Type ydbType;
-    if (typeInfo.GetTypeId() == NKikimr::NScheme::NTypeIds::Pg) {
-        auto typeDesc = typeInfo.GetPgTypeDesc();
-        auto* pg = ydbType.mutable_pg_type();
-        pg->set_type_name(NKikimr::NPg::PgTypeNameFromTypeDesc(typeDesc));
-        pg->set_oid(NKikimr::NPg::PgTypeIdFromTypeDesc(typeDesc));
-    } else {
-        auto& item = notNull
-            ? ydbType
-            : *ydbType.mutable_optional_type()->mutable_item();
-        //
-        // DECIMAL is PrimitiveType with (22,9) defaults in Scheme
-        // and separate (non-primitive) type everywhere else
-        //
-        // NKikimr::NScheme::NTypeIds::Decimal is omitted in public API intentionally
-        //
-        if (typeInfo.GetTypeId() == NKikimr::NScheme::NTypeIds::Decimal) {
-            auto* decimal = item.mutable_decimal_type();
-            decimal->set_precision(NKikimr::NScheme::DECIMAL_PRECISION);
-            decimal->set_scale(NKikimr::NScheme::DECIMAL_SCALE);
-        } else {
-            item.set_type_id((Ydb::Type::PrimitiveTypeId)typeInfo.GetTypeId());
-        }
-    }
-    return ydbType;
-}
-
 TExprNode::TPtr BuildSchemaFromMetadata(TPositionHandle pos, TExprContext& ctx, const TMap<TString, NYql::TKikimrColumnMetadata>& columns) {
     TVector<std::pair<TString, const NYql::TTypeAnnotationNode*>> typedColumns;
     typedColumns.reserve(columns.size());
     for (const auto& [n, c] : columns) {
-        NYdb::TTypeParser parser(NYdb::TType(CreateYdbType(c.TypeInfo, c.NotNull)));
+        Ydb::Type typeProto;
+        NKikimr::NScheme::ProtoFromTypeInfo(c.TypeInfo, typeProto, c.NotNull);
+        NYdb::TTypeParser parser((NYdb::TType(typeProto)));
```

Comment:
```suggestion
        NYdb::TTypeParser parser(NYdb::TType(typeProto));
```

Reply:
Current version is right. Without brakes there will be compilation error.

Reply:
So it's some cast? Maybe make it explicit? Or it's common practice for this code?

Reply:
Managed to simplify it removing cast at all
58372b3e159073df5e5b8286aa132cc16f8979ed